src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -5277,9 +5282,14 @@ instruct vsra16B_imm(vecX dst, vecX src, immI shift) %{
   format %{ "sshr    $dst, $src, $shift\t# vector (16B)" %}
   ins_encode %{
     int sh = (int)$shift$$constant;
-    if (sh >= 8) sh = 7;
-    __ sshr(as_FloatRegister($dst$$reg), __ T16B,
-           as_FloatRegister($src$$reg), sh);
+    if (sh == 0) {
```

Comment:
If src and dst are the same reg, no need to emit code. Or maybe c2 can even be improved to optimize this out?

Reply:
> If src and dst are the same reg, no need to emit code. 
If we want to do this enhancement, I think we need do it for left shifting and all SVE left/right shifting as well for completeness.
> Or maybe c2 can even be improved to optimize this (sh=0 case) out?
We can add code in `Ideal` to optimize it to ORR, but I'm not sure `orr` performs better than `shift` on other platforms.
Seems we have to created a generic new node to do `vector move` here.

Reply:
I think with proper optimization, no move is required. But I agree it's beyond the scope of this patch. I will have a look.